src/com/ms/silverking/cloud/dht/management/SKAdmin.java

```diff
@@ -247,1217 +269,1325 @@ public SKAdmin(SKGridConfiguration gc, SKAdminOptions options) throws IOExceptio
-        } else {
-            // non-escaped path is for generation of commands to run through e.g. zapp
-            reapPolicyOptions = "-reapPolicy " + ObjectDefParser2.toClassAndDefString(reapPolicy);
-        }
+    activeHostGroupNames = dhtConfig.getHostGroups();
+    Log.warning("hostGroupNames: ", CollectionUtil.toString(activeHostGroupNames));
+    hostGroupToClassVars = getHostGroupToClassVarsMap(dhtConfig);
+    Log.warning("hostGroupToClassVars: ", CollectionUtil.mapToString(hostGroupToClassVars));
+    Log.warning("ringConfig: ", ringConfig);
+    hostGroupTableName = ringConfig.getCloudConfiguration().getHostGroupTableName();
+    Log.warning("hostGroupTableName: ", hostGroupTableName);
+    hostGroupTable = getHostGroupTable(hostGroupTableName, dhtMC.getZooKeeper().getZKConfig());
+    Log.warning("HostGroupTable: " + hostGroupTable.toString());
+
+    passiveNodeHostGroupNames = dhtConfig.getPassiveNodeHostGroupsAsSet();
+    Log.warning("passiveNodeHostGroupNames: ", CollectionUtil.toString(passiveNodeHostGroupNames));
+
+    serverClassVars = getServerClassVars(thisServer, hostGroupTable, activeHostGroupNames, passiveNodeHostGroupNames,
+        hostGroupToClassVars);
+
+    reapPolicy = options.getReapPolicy();
+    generatedCmd = _generateNodeStartCommand(serverClassVars, options, reapPolicy, false);
+    Log.warning("Command generated for target " + thisServer);
+
+    System.out.println(generatedCmd);
+    return true;
+  }
 
-        return getJavaCmdStart(options, classVars, escaped)
-                +" "+ DHTNode.class.getCanonicalName()
-                +" " + reapPolicyOptions
-                +" -n "+ gc.getClientDHTConfiguration().getName()
-                +" -z "+ gc.getClientDHTConfiguration().getZKConfig()
-                +" -into "+ options.inactiveNodeTimeoutSeconds
-                ;
+  private String _generateNodeStartCommand(ClassVars classVars, SKAdminOptions options, ReapPolicy reapPolicy,
+      boolean escaped) {
+    String reapPolicyOptions;
+
+    if (escaped) {
+      reapPolicyOptions = "-reapPolicy " + escapeStr + ObjectDefParser2.toClassAndDefString(reapPolicy) + escapeStr;
+    } else {
+      // non-escaped path is for generation of commands to run through e.g. zapp
+      reapPolicyOptions = "-reapPolicy " + ObjectDefParser2.toClassAndDefString(reapPolicy);
     }
-    
-    private String getTaskset(SKAdminOptions options) {
-        if (options.pinToNICLocalCPUs != null) {
-            String    nic;
-            
-            nic = options.pinToNICLocalCPUs;
+
+    return getJavaCmdStart(options, classVars,
+        escaped) + " " + options.mainClass + " " + options.extraMainClassArgs + " " + reapPolicyOptions + " -n " + gc.getClientDHTConfiguration().getName() + " -z " + gc.getClientDHTConfiguration().getZKConfig() + " -into " + options.inactiveNodeTimeoutSeconds + (
+        options.overridePort >= 0 ?
+            String.format(" -port %s", options.overridePort) :
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: options.overridePort. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 566. Consider formatting this data with &#39;%d&#39; instead.